src/ex1/script.js

```diff
@@ -0,0 +1,165 @@
+const allTodosList = document.getElementById("all-todos-list");
+
+const unsorted = 0;
+const sortedAsc = 1;
+const sortedDesc = 2;
+let isListSorted = unsorted;
+
+const todosArray = ['Wash the dishes', 'Walk the dog', 'Water the flower', 'Feed the baby'];
+const inputTitle = document.getElementById("new-todo-title");
+const amountOfTodosInfo = document.getElementById("amount-info");
+
+const clearAllButton = document.getElementById("clear-all-button");
+const addTodoForm = document.getElementById("add-todo");
+const sortListButton = document.getElementById("sort-list-button");
+
+clearAllButton.addEventListener('click', onClearAllButtonClicked);
+addTodoForm.addEventListener('submit', onAddTodoFormSubmitted);
+sortListButton.addEventListener('click', onSortListButtonClicked);
+
+todosArray.forEach(todoText => {
+  addTodoItem(todoText, false);
+  displayFooterAndImage();
+});
+
+function addEventListenerForTodoTitle(listItem) {
+  const todoTitleDiv = listItem.getElementsByTagName("div")[0];
+  todoTitleDiv.addEventListener('click', ({target}) => {
+    onTodoTitleClicked(target);
+  });
+}
+
+function addEventListenerForDeleteButton(listItem) {
+  const deleteButton = listItem.getElementsByTagName("button")[0]
+  deleteButton.addEventListener('click', ({currentTarget}) => {
+    onDeleteButtonClicked(currentTarget);
+  });
+}
+
+function addTodoItem(todoText, animation) {
+  const listItem = createListElement(todoText, animation);
+  addEventListenerForTodoTitle(listItem);
+  addEventListenerForDeleteButton(listItem);
+  allTodosList.appendChild(listItem);
+  isListSorted = unsorted;
+}
+
+function createListElement(todoText, animation) {
+  const listItem = document.createElement("li");
+  if (animation) {
+    listItem.className = "add-item-animation";
+  }
+  setTimeout (() => { listItem.className = "existing-item";}, 700);
+  listItem.innerHTML = `<div class="todo-text">${todoText}</div>
+                        <button class="remove-todo-button"><i class="fa fa-trash"></i></button>`;
+  return listItem;
+}
+
+function displayFooterAndImage() {
+  displayButtonsAndAmount();
+  displayZeroImage();
+}
+
+function displayButtonsAndAmount() {
+  let tasks = "tasks";
+  if (todosArray.length === 1) {
+    tasks = "task";
+    displayElement('sort-list-button', false)
+    displayElement('clear-all-button', false)
+  } else {
+    displayElement('sort-list-button', true)
+    displayElement('clear-all-button', true)
+  }
+  amountOfTodosInfo.textContent = `${todosArray.length} pending ${tasks}`;
+}
+
+function displayZeroImage() {
+  if (todosArray.length === 0) {
+    displayElement('zero-todos-image', true)
+    displayElement('footer', false)
+  } else {
+    displayElement('zero-todos-image', false)
+    displayElement('footer', true)
+  }
+}
+
+function displayElement(elementId, toDisplay) {
```

Comment:
a bit confusing as the function name is display but its also used for hiding

Reply:
fixed and also added constant variables to call the function with DISPLAY or HIDE instead of true or false